### PR TITLE
Dev support concurrent runs of space tests

### PR DESCRIPTION
**Pull request by achilleasa**
## Description of change

This PR allows the EC2 space-based integration tests to run concurrently.

### CODE REVIEW

File: tests/suites/spaces_ec2/task.sh [hunk 1]
```diff
@@ -9,31 +9,61 @@ test_spaces_ec2() {
...
+    echo "==> Checking for stale NIC resources"
+    cleanup_stale_nics
+
+    echo "==> Setting up additional EC2 resources for tests"
+    hotplug_nic_id=$(setup_nic_for_space_tests)
+    echo "[+] created secondary NIC with ID ${hotplug_nic_id}"
+    function remove_nic() {
+      aws ec2 delete-network-interface --network-interface-id "$hotplug_nic_id"
+    }
+    trap remove_nic EXIT
```
Comment (SimonRichardson) [inline tests/suites/spaces_ec2/task.sh]: I think you remove the old trap here i.e. you can't have multiple traps for the same signal.

https://github.com/juju/juju/blob/develop/tests/main.sh#L248

File: tests/suites/spaces_ec2/task.sh [hunk 2]
```diff
@@ -9,31 +9,61 @@ test_spaces_ec2() {
...
+	# try to delete anything older in a best effort manner.
+	aws ec2 describe-network-interfaces --filter Name=description,Values="hot-pluggable NIC for space tests" \
+		| grep 'NetworkInterfaceId\|Value' \
+		| cut -d'"' -f4 \
+		| paste - - -d, \
+		| grep -v "$(date +'%Y-%m-%d')" \
+		| cut -d',' -f 1 \
+		| xargs -I % echo 'echo "[!] attempting to remove stale NIC % (will retry later if currently in-use)"; aws ec2 delete-network-interface --network-interface-id % 2>/dev/null' \
+		| sh \
+		|| true
```
Comment (SimonRichardson) [inline tests/suites/spaces_ec2/task.sh]: This makes sense...

**Comment by achilleasa** [comment]
!!build!!

File: tests/suites/spaces_ec2/juju_bind.sh [hunk 3]
```diff
@@ -9,8 +9,10 @@ run_juju_bind() {
     juju reload-spaces
     juju add-space isolated 172.31.254.0/24
 
-    ## Create machine
-    add_multi_nic_machine
+    # Note that due to the way that run_* funcs are executed, $1 holds the
+    # test name so the NIC ID is actually provided in $2
+    hotplug_nic_id=$2
```
Comment (SimonRichardson) [inline tests/suites/spaces_ec2/juju_bind.sh]: Did we not want to fix this?

Comment (achilleasa) [reply tests/suites/spaces_ec2/juju_bind.sh]: Yes but I thought it was less effort to simply follow the same pattern as the other tests which we already know that work as expected. Perhaps we can do this via a separate PR.

**Comment by SimonRichardson** [comment]
Awesome 👍

**Comment by achilleasa** [comment]
!!build!!

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**